src/controller/AutoCommissioner.cpp

```diff
@@ -777,17 +785,76 @@ CHIP_ERROR AutoCommissioner::CommissioningStepFinished(CHIP_ERROR err, Commissio
                     mParams.ClearICDStayActiveDurationMsec();
                 }
             }
+
+#if CHIP_DEVICE_CONFIG_ENABLE_JOINT_FABRIC
+            if (mParams.GetExecuteJCM().ValueOr(false) &&
+                (mDeviceCommissioningInfo.JFAdministratorFabricIndex != kUndefinedFabricIndex))
+            {
+                if (!mJFAdminRCAC.Alloc(mDeviceCommissioningInfo.JFAdminRCAC.size()))
+                {
+                    ChipLogError(Controller, "AutoCommissioner: cannot allocate memory for mJFAdminRCAC");
+                    return CHIP_ERROR_NO_MEMORY;
+                    ;
+                }
+                memcpy(mJFAdminRCAC.Get(), mDeviceCommissioningInfo.JFAdminRCAC.data(), mJFAdminRCAC.AllocatedSize());
+                mParams.SetJFAdminRCAC(ByteSpan(mJFAdminRCAC.Get(), mJFAdminRCAC.AllocatedSize()));
+
+                if (!mJFAdminICAC.Alloc(mDeviceCommissioningInfo.JFAdminICAC.size()))
```

Comment:
I assume that size is known to be nonzero here?

Reply:
mDeviceCommissioningInfo.JFAdminICAC.size() represents the size of the peer Admin's ICAC, as read in ParseJFAdministratorFabricInfo (_info.JFAdminICAC = nocStruct.icac.Value()_). I think the size could be 0 if the peer Admin is a rogue device, but the ICAC validity will be checked later in the process (as a side note, this is pretty similar to the flow that already exists for reportPAISpan from Autocommissioner.cpp).

Reply:
Added the check for 0 size.